src/Adapter/Feature/Repository/FeatureValueRepository.php

```diff
@@ -206,21 +228,34 @@ private function getFeatureValueLocalizedValues(int $featureValueId): array
     }
 
     /**
-     * @param ProductId $productId
      * @param array|null $filters
      *
      * @return QueryBuilder
      */
-    private function getFeatureValuesQueryBuilder(ProductId $productId, ?array $filters): QueryBuilder
+    private function getFeatureValuesQueryBuilder(?array $filters): QueryBuilder
     {
-        //@todo: filters are not handled.
         $qb = $this->connection->createQueryBuilder();
         $qb->from($this->dbPrefix . 'feature_value', 'fv')
             ->leftJoin('fv', $this->dbPrefix . 'feature_product', 'fp', 'fp.id_feature_value = fv.id_feature_value')
-            ->where('fp.id_product = :productId')
-            ->setParameter('productId', $productId->getValue())
         ;
 
+        foreach ($filters as $key => $value) {
+            switch ($key) {
+                case 'id_product':
+                    $qb
+                        ->andWhere('fp.id_product = :productId')
+                        ->setParameter('productId', (int) $value)
+                    ;
+                break;
+                default:
+                    $qb
```

Comment:
Should we use a secure list of authorized filters like we did for example in https://github.com/PrestaShop/PrestaShop/blob/develop/src/Core/Grid/Query/CmsPageCategoryQueryBuilder.php#L115 ?

Reply:
It was a very basic rough implementation I didn't put much thought into it 😅  But yes, your suggestion will definitely improve it